res/controllers/Hercules-DJ-Console-4-Mx-scripts.js

```diff
@@ -0,0 +1,622 @@
+        //Tell shift not to change state.
+        Hercules4Mx.shiftUsed = true;
+    }
+    switch (control) {
+        // 1 to 6 and 0x21 to 0x26 are the 6 left deck and right deck buttons respectively, non shifted.
+        // Loop buttons
+        case 0x01:
+        case 0x21: // K1, beatloop 1/ Loop in
+            if (value) {
+                //Loop in in "shift-pressed" mode, beatloop of 1 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "beatloop_0.125_toggle", 1);
+                } else {
+                    engine.setValue(group, "beatloop_0.5_toggle", 1);
+                }
+            }
+            break;
+        case 0x02:
+        case 0x22: // K2, beatloop 2/ Loop out
+            if (value) {
+                //Loop out in "shift-pressed" mode, beatloop of 2 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "beatloop_0.25_toggle", 1);
+                } else {
+                    engine.setValue(group, "beatloop_1_toggle", 1);
+                }
+            }
+            break;
+        case 0x03:
+        case 0x23: // K3, beatloop 4/ Reloop/Exit
+            if (value) {
+                //enable/disable loop in "shift-pressed" mode, beatloop of 2 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "reloop_exit", 1);
+                } else {
+                    engine.setValue(group, "beatloop_2_toggle", 1);
+                }
+            }
+            break;
+        case 0x04:
+        case 0x24: // K4, beatloop 8/ Reloop/Exit
+            if (value) {
+                //enable/disable loop in "shift-pressed" mode, beatloop of 4 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "reloop_exit", 1);
+                } else {
+                    engine.setValue(group, "beatloop_4_toggle", 1);
+                }
+            }
+            break;
+            // Reverse play buttons
+        case 0x05:
+        case 0x25: // K5, reverse play
+            engine.setValue(group, "reverse", (value) ? 1 : 0);
+            engine.setValue(group, "reverse", (value) ? 1 : 0);
+            break;
+        case 0x06:
+        case 0x26: // K6, reverse roll play
+            engine.setValue(group, "reverseroll", (value) ? 1 : 0);
+            engine.setValue(group, "reverse", (value) ? 1 : 0);
+            break;
+
+            // 7 to 0xC and 0x27 to 0x3C are the 6 left deck and right deck buttons respectively, with shift enabled.
+            // Hotcue buttons:
+        case 0x07:
+        case 0x27: // K7 Hotcue 1 activate/ clear
+            if (Hercules4Mx.shiftPressed) {
+                if (value) {
+                    engine.setValue(group, "hotcue_1_clear", 1);
+                }
+            } else engine.setValue(group, "hotcue_1_activate", (value) ? 1 : 0);
```

Comment:
> The only small glitch i found is that if i change the element on the sidebar and then go to files, the first up/down movement doesn't really move because the cursor is not placed anywhere and that first button press puts it at the beginning.

Yes, this is a kinda annoying little issue. A user just asked about that on the forum the other day: http://mixxx.org/forums/viewtopic.php?f=3&t=7780